src/misc/lv_fs.c

```diff
@@ -310,19 +309,19 @@ lv_fs_res_t lv_fs_write(lv_fs_file_t * file_p, const void * buf, uint32_t btw, u
         }
     }
 
-    uint32_t bw_tmp = 0;
+    size_t bw_tmp = 0;
     res = file_p->drv->write_cb(file_p->drv, file_p->file_d, buf, btw, &bw_tmp);
     if(bw != NULL) *bw = bw_tmp;
 
     if(file_p->drv->cache_size && res == LV_FS_RES_OK)
-        file_p->cache->file_position += bw_tmp;
+        file_p->cache->file_position += (uint32_t)bw_tmp;
```

Comment:
Both `file_position` and `bw_tmp` are of `size_t` type and do not need to be cast.